### PR TITLE
update license list

### PR DESCRIPTION
**Pull request by jspaaks**
**Related issues**

Refs:
- #454

**Describe the changes made in this pull request**

This PR updates the list of licenses as used in the schema to a newer version. I used https://github.com/citation-file-format/get-spdx-licenses to retrieve the new list.

**Review checklist**

- [x] Please check if the pull request is against the correct branch  
(format/schema/semantic documentation changes: `develop`; typos, meta files, etc.: `main`)
- [x] Please check if all changes are recorded in `CHANGELOG.md` and adapt if necessary.
- [x] Look at the diff
- [x] Please run tests locally.
```bash
cd $(mktemp -d --tmpdir cff.XXXXXX)
git clone https://github.com/citation-file-format/citation-file-format .
git checkout <branch>
python3 -m venv env
source env/bin/activate
pip install --upgrade pip wheel setuptools
pip install -r requirements.txt
pytest
```

### CODE REVIEW

**Comment by jspaaks** [comment]
Link checker fails because Wayback Machine links are resolving too slowly.

File: schema-guide.md [hunk 1]
```diff
@@ -1899,7 +1899,7 @@ authors:
     - `ZPL-2.0`
     - `ZPL-2.1`
 - **required**: N/A
-- **description**: [SPDX identifier](https://spdx.dev/ids/) for the license under which a work is made available. The list of identifiers originates from https://github.com/spdx/license-list-data/blob/bd8e963a41b13524b2ccb67f9335d2dd397c378e/json/licenses.json.
+- **description**: [SPDX identifier](https://spdx.dev/ids/) for the license under which a work is made available. The list of identifiers originates from https://raw.githubusercontent.com/spdx/license-list-data/e38c53a9c448a999f3ef772c0e019eb4dd2b0e2a/json/licenses.json.
```
Comment (sdruskat) [inline schema-guide.md:1902]: Much better, well spotted @jspaaks!

**Review by sdruskat — APPROVED** [review]
This one's good to go. Thanks, @jspaaks!

**Comment by jspaaks** [comment]
Thanks for reviewing! :cake:

**Pull request merged**